test/suites/integration/conversion_test.go

```diff
@@ -0,0 +1,132 @@
+					Template: karpv1beta1.NodeClaimTemplate{
+						Spec: karpv1beta1.NodeClaimSpec{
+							NodeClassRef: &karpv1beta1.NodeClassReference{
+								Name: "test-nodeclass",
+							},
+							Requirements: []karpv1beta1.NodeSelectorRequirementWithMinValues{
+								{
+									NodeSelectorRequirement: corev1.NodeSelectorRequirement{
+										Key:      corev1.LabelOSStable,
+										Operator: corev1.NodeSelectorOpIn,
+										Values:   []string{string(corev1.Linux)},
+									},
+								},
+								{
+									NodeSelectorRequirement: corev1.NodeSelectorRequirement{
+										Key:      karpv1.CapacityTypeLabelKey,
+										Operator: corev1.NodeSelectorOpIn,
+										Values:   []string{karpv1.CapacityTypeOnDemand},
+									},
+								},
+								{
+									NodeSelectorRequirement: corev1.NodeSelectorRequirement{
+										Key:      v1.LabelInstanceCategory,
+										Operator: corev1.NodeSelectorOpIn,
+										Values:   []string{"c", "m", "r"},
+									},
+								},
+								{
+									NodeSelectorRequirement: corev1.NodeSelectorRequirement{
+										Key:      v1.LabelInstanceGeneration,
+										Operator: corev1.NodeSelectorOpGt,
+										Values:   []string{"2"},
+									},
+								},
+								// Filter out a1 instance types, which are incompatible with AL2023 AMIs
+								{
+									NodeSelectorRequirement: corev1.NodeSelectorRequirement{
+										Key:      v1.LabelInstanceFamily,
+										Operator: corev1.NodeSelectorOpNotIn,
+										Values:   []string{"a1"},
+									},
+								},
+							},
+							Kubelet: &karpv1beta1.KubeletConfiguration{MaxPods: lo.ToPtr[int32](20)},
+						},
+					},
+				},
+			}
+
+			env.ExpectCreated(storedv1beta1NodePool.DeepCopy())
+			v1beta1NodePool := env.ExpectExists(storedv1beta1NodePool.DeepCopy()).(*karpv1beta1.NodePool)
+			Expect(v1beta1NodePool.Generation).To(BeNumerically("==", 1))
+
+			// Second apply of the same NodePool does not increase the generation
+			env.ExpectUpdated(storedv1beta1NodePool.DeepCopy())
+			v1beta1NodePool = env.ExpectExists(storedv1beta1NodePool.DeepCopy()).(*karpv1beta1.NodePool)
+			Expect(v1beta1NodePool.Generation).To(BeNumerically("==", 1))
+		})
+		It("should not update a metadata generation when the same resource is applied for the v1 APIs", func() {
+			env.ExpectCreated(nodePool.DeepCopy())
+			v1NodePool := env.ExpectExists(nodePool).(*karpv1.NodePool)
+			Expect(v1NodePool.Generation).To(BeNumerically("==", 1))
+
+			// Second apply of the same NodePool does not increase the generation
+			env.ExpectUpdated(nodePool.DeepCopy())
+			v1NodePool = env.ExpectExists(nodePool).(*karpv1.NodePool)
+			Expect(v1NodePool.Generation).To(BeNumerically("==", 1))
+		})
+	})
+	Context("EC2NodeClass", func() {
+		It("should not update a metadata generation when the same resource is applied for the v1beta1 APIs", func() {
+			// created v1beta1 resource
+			storedv1beta1nodeclass := test.BetaEC2NodeClass()
```

Comment:
Same question here for v1 EC2NodeClass

Reply:
What are the fields the are problematic? Are you referring to KubeletConfiguration? Those don't have any `resource.Quantity` types